Add a jenkins service override with TimeoutStartSec

Crola1702 opened the pull request:
When jenkins.service starts for the first time, it needs to execute all the scripts below `init.groovy.d`. The sum of all these scripts take a lot of time (specially `99-approved_signatures.groovy`).

The default timeout for the unit is 90s, but the execution takes more time than that (at least 2 minutes)

This PR adds a template for `jenkins.service.d/override.conf` which contains `TimeoutStartSec` defined as 180s (3 minutes)

Also, I removed a restart that was causing a `systemd unit start limit hit` error when restarting jenkins

Crola1702 committed:
Fix typo in gpg private-key

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Fix typo in gpg private-key (again)

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Revert "Fix typo in gpg private-key (again)"

See https://github.com/osrf/chef-osrf/pull/303

This reverts commit 3773a05cf89c1a426b3ca581b5a10b1e989d9f4d.

Crola1702 committed:
Add a jenkins service override with TimeoutStartSec

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Reduce timeout to 3 mins

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Remove restart of auth_strategy

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

claraberendsen commented on recipes/jenkins.rb:
```diff
@@ -92,6 +92,22 @@
+
+template '/etc/systemd/system/jenkins.service.d/override.conf' do
+  source 'jenkins/jenkins-service-override.conf.erb'
+  owner node['jenkins']['master']['user']
+  group node['jenkins']['master']['group']
```
Should this be owned by root as the folder as well? Or is there a reason for this being owned by the jenkins user?

nuclearsandwich replied:
Yes, it is read by systemd and not the jenkins user.

nuclearsandwich commented on recipes/jenkins.rb:
```diff
@@ -92,6 +92,22 @@
+  group 'root'
+end
+
+template '/etc/systemd/system/jenkins.service.d/override.conf' do
```
This will collide with the file created in the jenkins cookbook: https://github.com/ros-infrastructure/cookbook-jenkins/blob/9435bd9121d0c38b745a5548b4f1ea8365c7f43e/recipes/jenkins.rb#L67-L73

systemd can actually process multiple override files. Any *.conf file in the directory. So we could create a single `500-timeout-start.conf` with fixed content and avoid using a template file here entirely.

nuclearsandwich commented on recipes/jenkins.rb:
```diff
@@ -92,6 +92,22 @@
+  variables Hash[
+    timeout_start_sec: 180
+  ]
+end
```
In order to take effect this needs to trigger a systemd daemon-reload: https://github.com/ros-infrastructure/cookbook-jenkins/blob/9435bd9121d0c38b745a5548b4f1ea8365c7f43e/recipes/jenkins.rb#L72-L73

nuclearsandwich commented on recipes/jenkins.rb:
```diff
@@ -207,12 +223,6 @@ def jenkins = Jenkins.getInstance()
-    action :restart
-  end
-
   # Aggregate permissions to assign to each user with a groovy script.
```
This seems like treatment of a symptom rather than the underlying problem.

Changing one or both of TimeoutStopSec or RestartSec seems more appropriate since normal cookbook convergence shouldn't cause failures starting during configuration.

Crola1702 replied:
I used RestartSec on the coookbook_file

Crola1702 committed:
Use cookbook_file with timeouts override configs

Set 500-timeout.conf cookbook_file with TimeoutStartSec and RestartSec options

TimeoutStartSec is set to prevent systemd killing jenkins service when it is trying to run all the scripts below init.groovy.d/
RestartSec option is set to prevent the start limit hit when running the cookbook

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

Crola1702 committed:
Remove jenkins service override template

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>

claraberendsen commented on recipes/jenkins.rb:
```diff
@@ -92,6 +92,23 @@
 
 include_recipe 'jenkins::jenkins'
 
+# Increase timeout of jenkins systemd unit
```
(nit) It would be good to add rationale behind the increase of the timeout in this comment to have a pointer as to why this was added.

Crola1702 committed:
Add rationale behind timeout increase

Signed-off-by: Crola1702 <cristobal.arroyo@ekumenlabs.com>
